owncloudApp/src/main/java/com/owncloud/android/providers/DocumentsStorageProvider.kt

```diff
@@ -0,0 +1,300 @@
+        }
+
+        return null
+    }
+
+    private fun updateCurrentStorageManagerIfNeeded(docId: Long) {
+        if (rootIdToStorageManager.isEmpty()) {
+            initiateStorageMap()
+        }
+        if (currentStorageManager == null || (
+                    rootIdToStorageManager.containsKey(docId) &&
+                            currentStorageManager !== rootIdToStorageManager[docId])
+        ) {
+            currentStorageManager = rootIdToStorageManager[docId]
+        }
+    }
+
+    private fun updateCurrentStorageManagerIfNeeded(rootId: String) {
+        if (rootIdToStorageManager.isEmpty()) {
+            return
+        }
+        for (data in rootIdToStorageManager.values) {
+            if (data.account.name == rootId) {
+                currentStorageManager = data
+            }
+        }
+    }
+
+    private fun initiateStorageMap() {
+        val contentResolver = context!!.contentResolver
+
+        for (account in AccountUtils.getAccounts(context)) {
+            val storageManager = FileDataStorageManager(context, account, contentResolver)
+            val rootDir = storageManager.getFileByPath(OCFile.ROOT_PATH)
+            rootIdToStorageManager[rootDir!!.fileId] = storageManager
+        }
+    }
+
+    private fun syncDirectoryWithServer(parentDocumentId: String) {
+        val folderId = parentDocumentId.toLong()
+        val refreshFolderOperation = RefreshFolderOperation(
+            currentStorageManager?.getFileById(folderId),
+            false,
+            false,
+            currentStorageManager?.account,
+            context
+        )
+        refreshFolderOperation.syncVersionAndProfileEnabled(false)
+
+        val contentResolver = context!!.contentResolver
+
+        val thread = Thread {
+            refreshFolderOperation.execute(currentStorageManager, context)
+            contentResolver.notifyChange(toNotifyUri(toUri(parentDocumentId)), null)
+        }
+        thread.start()
+    }
+
+    private fun waitOrGetCancelled(cancellationSignal: CancellationSignal?): Boolean {
+        try {
+            Thread.sleep(1000)
+        } catch (e: InterruptedException) {
+            return false
+        }
+
+        return cancellationSignal == null || !cancellationSignal.isCanceled
+    }
+
+    private fun findFiles(root: OCFile, query: String): Vector<OCFile> {
+        val result = Vector<OCFile>()
+        for (f in currentStorageManager!!.getFolderContent(root,false)) {
+            if (f.isFolder) {
+                result.addAll(findFiles(f, query))
+            } else {
+                if (f.fileName.contains(query)) {
+                    result.add(f)
+                }
+            }
+        }
+        return result
+    }
+
+    private fun toNotifyUri(uri: Uri): Uri {
+        return DocumentsContract.buildDocumentUri(
+            documentsProviderAuthority, uri.toString()
+        )
+    }
+
+    private fun toUri(documentId: String): Uri {
```

Comment:
And rewrite this function in just one line 

`private fun toUri(documentId: String): Uri =  Uri.parse(documentId)`